src/shared/bus-util.c

```diff
@@ -578,6 +580,59 @@ int bus_reply_pair_array(sd_bus_message *m, char **l) {
         return sd_bus_send(NULL, reply, NULL);
 }
 
+static int method_dump_memory_state_by_fd(sd_bus_message *message, void *userdata, sd_bus_error *ret_error) {
+        _cleanup_free_ char *dump = NULL; /* keep this above dump_file, so that it's freed after */
+        _cleanup_fclose_ FILE *dump_file = NULL;
+        _cleanup_close_ int fd = -EBADF;
+        size_t dump_size;
+        int r;
+
+        assert(message);
+
+        dump_file = open_memstream(&dump, &dump_size);
+        if (!dump_file)
+                return -ENOMEM;
+
+        r = RET_NERRNO(malloc_info(/* options= */ 0, dump_file));
+        if (r < 0)
+                return r;
+
+        dump_file = safe_fclose(dump_file);
+
+        fd = acquire_data_fd(dump, dump_size, 0);
+        if (fd < 0)
+                return fd;
+
+        r = sd_bus_reply_method_return(message, "h", fd);
+        if (r < 0)
+                return r;
+
+        return 1; /* Stop further processing */
+}
+
+/* The default install callback will fail and disconnect the bus if it cannot register the match, but this
+ * is only a debug method, we definitely don't want to fail in case there's some permission issue. */
+static int dummy_install_callback(sd_bus_message *message, void *userdata, sd_bus_error *ret_error) {
+        return 1;
+}
+
+int bus_register_malloc_status(sd_bus *bus) {
+        int r;
+
+        assert(bus);
+
+        r = sd_bus_add_match_async(bus,
+                                   NULL,
+                                   "type='method_call',interface='org.freedesktop.MallocStatus1',path='/org/freedesktop/MallocStatus1',member='GetMallocInfo'",
+                                   method_dump_memory_state_by_fd,
+                                   dummy_install_callback,
+                                   NULL);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to subscribe to GetMallocInfo() calls on MallocStatus1 interface: %m");
```

Comment:
hmm, in light of the SIGRTMIN+18 stuff #26393 will add I think we should give this dbus interface a slightly more generic name, so that a "MemoryTrim()" method could later be added nicely, which would not be so much about "status".

Hence maybe "org.freedesktop.MemoryAllocation1"?